plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/BasePlugin.java

```diff
@@ -65,21 +70,57 @@
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
+import javafx.beans.InvalidationListener;
+import javafx.beans.property.*;
+import javafx.beans.value.ChangeListener;
 
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.prefs.Preferences;
 
 @Description(
     group = "edu.wpi.first.shuffleboard",
     name = "Base",
-    version = "1.2.0",
+    version = "1.3.0",
     summary = "Defines all the WPILib data types and stock widgets"
 )
 @SuppressWarnings("PMD.CouplingBetweenObjects")
 public class BasePlugin extends Plugin {
+  private final Preferences preferences = Preferences.userNodeForPackage(getClass());
+
+  private final IntegerProperty graphUpdateRate = new SimpleIntegerProperty(this, "graphUpdateRate", 10);
+  private final InvalidationListener graphSaver = __ -> PreferencesUtils.save(graphUpdateRate, preferences);
+
+  // only written to one thread at a time
+  private volatile ScheduledFuture<?> currentFuture;
```

Comment:
Maybe track this somewhere else? It doesn't make sense for the plugin to have to manage custom logic for a widget's configuration.

Reply:
The way it works currently every graph updates at the same rate. It's a holdover from the previous impl.

Reply:
I'm talking more from a code organization standpoint. Its better to encapsulate the graph widget updater logic in its own class instead of managing it in the base plugin